src/gui/owncloudgui.cpp

```diff
@@ -1063,35 +1063,21 @@ void ownCloudGui::slotShowShareDialog(const QString &sharePath, const QString &l
             maxSharingPermissions = SharePermission(0);
         }
 
-
-        ShareDialog *w = nullptr;
-        if (_shareDialogs.contains(localPath) && _shareDialogs[localPath]) {
-            qCInfo(lcApplication) << "Raising share dialog" << sharePath << localPath;
-            w = _shareDialogs[localPath];
+        if (_shareDialog && _shareDialog->localPath() == localPath) {
```

Comment:
I guess we should return in that case, not sure what happens when addModalLegacyDialog is called multiple times with the same widget.